File: Javascript/Q01.js
/**
 * Coding Interview Question #01
 */

for (var i = 0; i < 3; i++) {
  setTimeout(() => {
    console.log(`With var = ${i}`);
  }, 1000 + i);
}

// output: 3 3 3

for (let j = 0; j < 3; j++) {
  setTimeout(() => {
    console.log(`With let = ${j}`);
  }, 1000 + j);
}

// output: 0 1 2

File: Python/Q01.py
#//usr/bin/env python3

# ########################################
# Coding Interview Question #01
# Cool function feature in Python:
# Function as a return value
# ########################################

def animal(name: str):
    print(name)
    return animal

animal('Dog')('Cat')('Horse')('Cow') #Output: Dog Cat Horse Cow
